### ISSUE TITLE
detect/report drift: compare desired with actual state

### ISSUE THREAD

**Issue by patrikbeno** [databus23/helm-diff#508]
Compare deployed helm release with actual live/runtime manifests, and report changes (drift).

This is a missing feature at the moment (v3.8.1), helm-diff only works with releases/revisions.

I guess this task would require fetching actual manifest referenced by helm release/revision and compare desired with actual state, and helm-diff is very close to supporting this.

**Comment by gustavobini** [comment]
I might have misunderstood this issue, but is this not what it does right now when doing an `upgrade --dry-run`?

**Comment by patrikbeno** [comment]
@gustavobini , no. upgrade --dry-run shows all manifests in its TO-BE state. This is about AS-IS state compared to desired helm revision, and showing only DIFF.

**Comment by patrikbeno** [comment]
The use case I had in mind is this:

```
helm install my-release .

# manual edit some resource deployed by my-release
kubectl edit ... 

# this compares actual runtime manifests with revision 1 of my-release
# in this example, it shows my edit I made in previous kubectl-edit
helm diff revision my-release runtime 1 
```

`runtime` may be a keyword here; since revisions are integers, this keyword would tell helm-diff to take manifests not from a specific revision but from actual runtime.

**Comment by itaispiegel** [comment]
My team also needs this feature so I'd like to thumbs up this issue 👍🏻

**Comment by itaispiegel** [comment]
Just looked into it and I think this is solved by using the `--three-way-merge` flag. At first I didn't understand the meaning of this flag, but then I found [this](https://helm.sh/docs/faq/changes_since_helm2/#improved-upgrade-strategy-3-way-strategic-merge-patches) documentation.

**Comment by patrikbeno** [comment]
> Just looked into it and I think this is solved by using the `--three-way-merge` flag. At first I didn't understand the meaning of this flag, but then I found [this](https://helm.sh/docs/faq/changes_since_helm2/#improved-upgrade-strategy-3-way-strategic-merge-patches) documentation.

`--three-way-merge` looks promising, but `helm-diff-upgrade` is missing `--force`
